src/app/shared/profile-image/profile-image.page.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, Input, OnInit, EventEmitter, Output, ViewChild, ElementRef } from '@angular/core';
+import { ToastService } from 'src/app/services/toast/toast.service';
+import { UtilService } from 'src/app/services/util/util.service';
+
+@Component({
+  selector: 'app-profile-image',
+  templateUrl: './profile-image.page.html',
+  styleUrls: ['./profile-image.page.scss'],
+})
+export class ProfileImagePage implements OnInit {
+  @ViewChild('fileUpload') fileUpload!: ElementRef;
+  @Input() profileImageData: any;
+  @Input() showProfileDetails: any;
+  @Input() username: any;
+  @Input() uploadImage: boolean = false;
+  @Output() imageUploadEvent = new EventEmitter();
+  @Output() imageRemoveEvent = new EventEmitter();
+  isMobile = this.utilService.isMobile();
+
+  constructor(
+    private toast: ToastService,
+    private utilService: UtilService,
+  ) { }
+
+  ngOnInit() { }
+
+  clearFileInput() {
+    if (this.fileUpload) {
+      this.fileUpload.nativeElement.value = '';
+    }
+  }
+
+  async uploadPhoto(source: string) {
+    switch (source) {
+      case 'CAMERA':
+        this.fileUpload.nativeElement.setAttribute('capture', 'environment');
```

Comment:
See somethings are common here, see if that can be outsourced to a function and in the whole logic as well